bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/OutputSaasMappingService.kt

```diff
@@ -23,21 +23,98 @@ import mu.KotlinLogging
 import org.eclipse.tractusx.bpdm.common.dto.saas.AugmentedBusinessPartnerResponseSaas
 import org.eclipse.tractusx.bpdm.common.dto.saas.BusinessPartnerSaas
 import org.eclipse.tractusx.bpdm.common.dto.saas.SharingStatusSaas
+import org.eclipse.tractusx.bpdm.common.dto.saas.isError
 import org.eclipse.tractusx.bpdm.common.service.SaasMappings
 import org.eclipse.tractusx.bpdm.gate.config.SaasConfigProperties
 import org.eclipse.tractusx.bpdm.gate.dto.response.ErrorInfo
 import org.eclipse.tractusx.bpdm.gate.exception.BusinessPartnerOutputError
 import org.eclipse.tractusx.bpdm.gate.filterNotNullKeys
+import org.eclipse.tractusx.bpdm.gate.model.BusinessPartnerSaaSWithBpn
+import org.eclipse.tractusx.bpdm.gate.model.BusinessPartnerSaaSWithExternalId
+import org.eclipse.tractusx.bpdm.gate.model.SharingStatusEvaluationResult
 import org.springframework.stereotype.Service
 import java.time.LocalDateTime
 
 @Service
 class OutputSaasMappingService(
     private val saasConfigProperties: SaasConfigProperties,
+    private val saasClient: SaasClient
 ) {
 
     private val logger = KotlinLogging.logger { }
 
+    fun mapWithLocalBpn(partners: Collection<BusinessPartnerSaas>): Collection<BusinessPartnerSaaSWithBpn> {
+        val partnersWithExternalId = mapWithExternalId(partners)
+        val externalIds = partnersWithExternalId.map { it.externalId }
+
+        val augmentedPartnerResponse = saasClient.getAugmentedLegalEntities(externalIds = externalIds)
```

Comment:
This function is used for LegalEntities, Sites and Addresses. 
Isn't it a problem to use `getAugmentedLegalEntities()` which filters for the type "legal entity"?